sapi/cgi/cgi_main.c

```diff
@@ -1468,7 +1468,7 @@ void fastcgi_cleanup(int signal)
 	if (parent && parent_waiting) {
 		exit_signal = 1;
 	} else {
-		exit(0);
+		_exit(0);
```

Comment:
Out of curiosity, is there a specific motivation for using `_exit`? I.e. why should `atexit`/`on_exit` handlers not be called?

Reply:
Yes, exit flushes stdio buffers, you cannot do that on signal handlers. stdio is the prime example of an interface unsafe to use in signal handlers.
Likelwise you cannot call *printf but since it is guarded as a debug statement I left it as is.

Reply:
The gory details are mostly explained here https://man7.org/linux/man-pages/man7/signal-safety.7.html

Reply:
@crrodriguez Thank you for the explanation! I suppose the `fprintf` is also unsound then, but I suppose nobody compiles with `-DDEBUG_FASTCGI`.

Reply:
> @crrodriguez Thank you for the explanation! I suppose the `fprintf` is also unsound then, but I doubt anybody compiles with `-DDEBUG_FASTCGI`.

Yes, that's why I leave it there, assuming only whoever wrote this code uses this flag to debug and will know what to do if the program crashes around there.